### PR TITLE
Enable ispn cache

### PR DESCRIPTION
**Pull request by rigazilla**
Integrated Infinispan cache services
updated the demo
added readme

replaced Json serializer with messagepack

### CODE REVIEW

File: src/Infinispan.Hotrod.EFCaching.csproj [hunk 1]
```diff
@@ -12,9 +12,11 @@
     <SignAssembly>false</SignAssembly>
     <PackageReleaseNotes />
   </PropertyGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\..\EFCoreSecondLevelCacheInterceptor\src\EFCoreSecondLevelCacheInterceptor\EFCoreSecondLevelCacheInterceptor.csproj" />
+  </ItemGroup>
```
Comment (NeerajGartia21) [inline src/Infinispan.Hotrod.EFCaching.csproj:17]: I think we need to refer the `EFCoreSecondLevelCacheInterceptor` package here.

File: example/EFCoreBasicCaching/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Infinispan second level cache for EF Core
+
+## Requirements
+- dotnet 7.0
+- qslite
```
Comment (NeerajGartia21) [inline example/EFCoreBasicCaching/README.md:5]: I think it should be `sqlite`.

File: src/EFCoreSecondLevelCacheInterceptor/EFInfinispanCacheServiceProvider.cs [hunk 3]
```diff
@@ -27,48 +34,158 @@ public void ClearAllCachedEntries()
...
+                {
+                    replaced = depCache.PutIfAbsent(rootCacheKey, new HashSet<EFCacheKey>() { cacheKey }).Result == null;
+                    continue;
+                }
+                old.Value.Add(cacheKey);
+                replaced = depCache.ReplaceWithVersion(rootCacheKey, old.Value, old.Version).Result;
+            }
+        }
+
+        var lifeSpan = cachePolicy.CacheTimeout;
```
Comment (NeerajGartia21) [inline src/EFCoreSecondLevelCacheInterceptor/EFInfinispanCacheServiceProvider.cs:61]: var `lifeSpan` is not used anywhere. Either we can omit it or use it below.

File: src/EFCoreSecondLevelCacheInterceptor/EFInfinispanCacheServiceProvider.cs [hunk 4]
```diff
@@ -12,13 +17,15 @@ public class EFInfinispanCacheServiceProvider : IEFCacheServiceProvider
 {
     InfinispanDG dg = new InfinispanDG();
     Cache<EFCacheKey, EFCachedData> cache;
+    Cache<string, HashSet<EFCacheKey>> depCache;
     public EFInfinispanCacheServiceProvider()
     {
         // Use a non-authenticated non-encrypted cluster;
         dg.AddHost("127.0.0.1", 11222);
```
Comment (NeerajGartia21) [inline src/EFCoreSecondLevelCacheInterceptor/EFInfinispanCacheServiceProvider.cs:24]: Is there any specific reason to use non-authenticated cluster ?

**Review by NeerajGartia21 — COMMENTED** [review]
The Implementation looks good to me !

File: src/EFCoreSecondLevelCacheInterceptor/EFInfinispanCacheServiceProvider.cs (hunk 4 above)
Comment (rigazilla) [reply src/EFCoreSecondLevelCacheInterceptor/EFInfinispanCacheServiceProvider.cs:24]: just to keep things a simple as possible for a POC.
All the configuration part is actually missing currently and needs a plan. Authentication will be there

**Review by NeerajGartia21 — APPROVED** [review]

**Comment by rigazilla** [comment]
thanks for the review @NeerajGartia21 !

**Pull request merged**